### ISSUE TITLE
CPU exhausted even with plenty available

### ISSUE THREAD

**Issue by meyerbro** [hashicorp/nomad#23314]
### Nomad version
1.8.0

### Operating system and Environment details
Amazon Linux 2023

### Issue
More than 4000 Mhz available on each node but can't scale extra 1000 Mhz alloc at all

### Reproduction steps
Run on EC2 (AL2023) with or without dmidecode and try to fit 90% CPU of any nomad client, it will stop at around 60% usage.
My CPU is around 10150Mhz in AWS for each node and I only reserve 500Mhz and no cpu cores for the nomad client.
Funny enough, if I set cpu_total_compute to 10000Mhz I can fit 9000Mhz without issues!
I have 7 nodes and I can fit 63 1000Mhz nginx allocs in previous versions of Nomad but can't without hacking cpu_total_compute to the same amount of Mhz I have on those nodes...
My nodes in AWS are "r5a.xlarge".

#### Expected Result
Fill 90% of the CPU, just like previous versions.

#### Actual Result
cpu exausted in x nodes

### Job file
```
job "job" {
  type = "service"
  datacenters = [REDACTED]
  namespace = "PLATFORM"
  group "group" {
    count = 63
    task "nginx" {
      resources {
        cpu    = 1000
        memory = 1024
      }
      driver = "docker"
      config {
        image = "nginx"
      }
    }
  }
}
```

**Comment by jrasell** [comment]
Hi @meyerbro and thanks for raising this issue. Is it possible to add additional detail of the error(s) you are seeing? It would be useful to have the output from when you try and register a job that should be placed that doesn't along with the node status showing the available resources.

**Comment by meyerbro** [comment]
Hi @jrasell, I just get "Resources exhausted on 7 nodes" and "Dimension "cpu" exhausted on 7 nodes". And in the UI I see placement failures for 21 unplaced allocations.

**Comment by tgross** [comment]
This looks to be the same issue as https://github.com/hashicorp/nomad/issues/23371; I'm working to get some eyes on this. (Internal ref https://hashicorp.atlassian.net/browse/NET-10107)

**Comment by pkazmierczak** [comment]
hey @meyerbro, the patch landed in main and will be released in 1.8.2. Thanks again for reporting!

**Comment by meyerbro** [comment]
Thank you all!
